File: src/types/chemicalTestType.ts
import { LocationType } from './locationType';
import { ChemicalTestKeyType } from '../enums/chemicalTestKeyType';
import { ValueType } from './valueType';

export type ChemicalTestType = {
  [ChemicalTestKeyType.Id]: string;
  [ChemicalTestKeyType.CarbonationDepth]: number;
  [ChemicalTestKeyType.ChlorideContent]: number;
  [ChemicalTestKeyType.AlkaliReactivity]: number;
  [ChemicalTestKeyType.Date]: string;
  [ChemicalTestKeyType.UserId]: string;
  [ChemicalTestKeyType.Location]: LocationType;
};

export const ChemicalTestValueMap: Record<ChemicalTestKeyType, ValueType> = {
  [ChemicalTestKeyType.Id]: 'string',
  [ChemicalTestKeyType.CarbonationDepth]: 'number',
  [ChemicalTestKeyType.ChlorideContent]: 'number',
  [ChemicalTestKeyType.AlkaliReactivity]: 'number',
  [ChemicalTestKeyType.Date]: 'string',
  [ChemicalTestKeyType.UserId]: 'string',
  [ChemicalTestKeyType.Location]: 'LocationType'
};

File: backend/slab2reuse/src/main/java/circrete/slab2reuse/model/ChemicalTestType.java
package circrete.slab2reuse.model;

import jakarta.persistence.*;

@Entity
public class ChemicalTestType {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private double carbonationDepth;
    private double chlorideContent;
    private double alkaliReactivity;
    private String date;
    private String userId;

    @OneToOne(cascade = CascadeType.ALL)
    private LocationType location;

    // Getters and Setters
    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public double getCarbonationDepth() {
        return carbonationDepth;
    }

    public void setCarbonationDepth(double carbonationDepth) {
        this.carbonationDepth = carbonationDepth;
    }

    public double getChlorideContent() {
        return chlorideContent;
    }

    public void setChlorideContent(double chlorideContent) {
        this.chlorideContent = chlorideContent;
    }

    public double getAlkaliReactivity() {
        return alkaliReactivity;
    }

    public void setAlkaliReactivity(double alkaliReactivity) {
        this.alkaliReactivity = alkaliReactivity;
    }

    public String getDate() {
        return date;
    }

    public void setDate(String date) {
        this.date = date;
    }

    public String getUserId() {
        return userId;
    }

    public void setUserId(String userId) {
        this.userId = userId;
    }

    public LocationType getLocation() {
        return location;
    }

    public void setLocation(LocationType location) {
        this.location = location;
    }
}